src/plugins/intel_cpu/src/emitters/snippets/x64/jit_parallel_loop_emitters.cpp

```diff
@@ -0,0 +1,417 @@
+#include "openvino/core/except.hpp"
+#include "openvino/core/type.hpp"
+#include "snippets/emitter.hpp"
+#include "snippets/kernel_executor_table.hpp"
+#include "snippets/lowered/expression.hpp"
+#include "snippets/op/loop.hpp"
+#include "snippets/utils/utils.hpp"
+
+using namespace Xbyak;
+using namespace dnnl::impl::cpu::x64;
+
+namespace ov::intel_cpu {
+
+// TODO: unify with jit_loop_emitters
+namespace {
+class jit_aux_gpr_holder {
+public:
+    jit_aux_gpr_holder(dnnl::impl::cpu::x64::jit_generator_t* host,
+                       std::vector<size_t>& pool_gpr_idxs,
+                       const std::vector<size_t>& used_gpr_idxs)
+        : m_h(host),
+          m_pool_gpr_idxs(pool_gpr_idxs) {
+        // If the pool is empty, let's manualy allocate the gpr and push original vlaue on stack
+        if (m_pool_gpr_idxs.empty()) {
+            m_aux_gpr_idx = ov::intel_cpu::utils::get_aux_gpr(used_gpr_idxs);
+            m_is_preserved = true;
+            m_h->push(m_aux_gpr_idx);
+        } else {
+            m_aux_gpr_idx = Reg64(static_cast<int>(m_pool_gpr_idxs.back()));
+            m_pool_gpr_idxs.pop_back();
+        }
+    }
+
+    ~jit_aux_gpr_holder() {
+        if (m_is_preserved) {
+            m_h->pop(m_aux_gpr_idx);
+        } else {
+            m_pool_gpr_idxs.push_back(m_aux_gpr_idx.getIdx());
+        }
+    }
+
+    [[nodiscard]] const Reg64& get_reg() const {
+        return m_aux_gpr_idx;
+    }
+
+private:
+    dnnl::impl::cpu::x64::jit_generator_t* m_h;
+    std::vector<size_t>& m_pool_gpr_idxs;
+    Reg64 m_aux_gpr_idx;
+    bool m_is_preserved = false;
+};
+}  // namespace
+
+jit_parallel_loop_base_emitter::jit_parallel_loop_base_emitter(jit_generator_t* h,
+                                                               cpu_isa_t isa,
+                                                               const ov::snippets::lowered::ExpressionPtr& expr)
+    : jit_binary_call_emitter(h, isa, expr->get_live_regs()) {
+    in_out_type_ = emitter_in_out_map::gpr_to_gpr;
+    std::shared_ptr<snippets::op::LoopEnd> loop_end;
+    std::vector<snippets::Reg> loop_end_input_regs;
+    if (auto loop_begin = ov::as_type_ptr<snippets::op::LoopBegin>(expr->get_node())) {
+        loop_end = loop_begin->get_loop_end();
+        // todo: A long-term solution would be to introduce loop expressions so LoopBeginExpr->get_loop_end() would
+        // return LoopEndExpr directly
+        const auto& consumers = expr->get_output_port_connector(expr->get_output_count() - 1)->get_consumers();
+        OV_CPU_JIT_EMITTER_ASSERT(!consumers.empty(), "LoopBegin must have LoopEnd as the last consumer");
+        const auto& loop_end_expr = consumers.rbegin()->get_expr();
+        OV_CPU_JIT_EMITTER_ASSERT(loop_end_expr && loop_end_expr->get_node() == loop_end,
+                                  "Failed to find valid LoopEnd expression");
+        loop_end_input_regs = loop_end_expr->get_reg_info().first;
+    } else {
```

Comment:
Could you please add the branch  `if (loop_end)` here and else - throw exception?